effectsize_exemplar_within.Rmd

```diff
@@ -0,0 +1,137 @@
+
+For those accustomed to Cohen's *d*, it's important to be aware that $\eta_{G}^{2}$ is typically smaller, with a Cohen's d of 0.2 being equivalent to a $\eta_{G}^{2}$ of around 0.02. Also, the actual number has little meaning beyond its scale relative to other effects. 
+
+```{r within-anova}
+results = afex::aov_ez(
+  data = data, 
+  id = 'id', # subject id column
+  dv = 'response_time', # dependent variable
+  within = c('layout', 'size', 'color'), # within-subject independent variables
+  between = NULL ,# between-subject independent variables
+  anova_table = list(es = 'ges') # effect size = generalized eta squared
+)
+```
+
+*Note: the warning indicates that the aov_ez function automatically collapses repetitions into a mean, which may be a problem if an experiment is not fully counterbalanced. This example, however, has every subject running in every combination of conditions, so simple collapsing is the correct procedure.*
+
+```{r  within-anova-cleanup}
+anovaResults <- 
+  results$anova_table %>% 
+  rownames_to_column('effect') %>%  # put effect names in a column
+  select(-`Pr(>F)`) # no need to show p-values
+  
+anovaResults %>% 
+  tidy() %>% 
+  rename(F = statistic)
+```
+
+*Note that the fractional degrees of freedom result from a Greenhousse-Geisser sphericity correction.*
+
+```
+TODO: Boostrapped 95%CIs for effect sizes
+Pro: people should
+Con: would make the guide even longer
+Maybe push into another guideline?
+```
+
+### Reporting the results
+
+Looking at the `F` and `ges` (generalized eta squared) columns, there are clear main effects for `layout` and `size` and an interaction between `layout` and `size`. However `color` and the other 2-way and 3-way interactions show only negligeable effects.
+
+```{r within-format, include=FALSE}
+# format the anova results for a report, and trim to 3 significant digits
+formatGES <- function(anovaTable, effectName) {
+  cutoff = 0.01
+  row = (1:nrow(anovaTable))[anovaTable$effect == effectName]
+  return(paste0(
+    'F~',
+    signif(anovaTable[row, 'num Df'], 3), ',',
+    signif(anovaTable[row, 'den Df'], 3), '~ = ',
+    signif(anovaTable[row, 'F'], 3), ', $\\eta_{G}^{2}$ = ',
+    signif(anovaTable[row, 'ges'], 3)
+  ))
+}
+```
+
+
+ - **layout:** `r formatGES(anovaResults, 'layout')`
+ - **size:** `r formatGES(anovaResults, 'size')`
+ - **layout** × **size:** `r formatGES(anovaResults, 'layout:size')`
+ - **color** did not have a substantive effect (`r formatGES(anovaResults, 'color')`)
+
+Report any interaction for which there is reason to believe an effect could occur. Otherwise, you can simply state that other 2-way and 3-way interactions did not have substantive effect sizes. However, when in doubt, report everything!
```

Comment:
@steveharoz I'm not sure I agree with this exactly. This smells a little like when people don't report non-significant tests. I'd rather a recommendation like, if you ran something, either report it, or if you don't because it wasn't substantive, include the estimates in an appendix or supplement and refer the reader there.